### PR TITLE
Add cleaning

### PR DESCRIPTION
**Pull request by drew-sinha**
In this push, remove_excluded_positions builds a "excluded_positions" subfolder with parallel structure to standard experiment folders. The docstring has some extensive elaboration on this.

### CODE REVIEW

File: elegant/clean_timepoint_data.py [hunk 1]
```diff
@@ -113,11 +122,30 @@ def remove_excluded_positions(experiment_root,dry_run=False):
     annotations = load_data.read_annotations(experiment_root)
     good_annotations = load_data.filter_annotations(annotations, load_data.filter_excluded)
     excluded_positions = sorted(set(annotations.keys()).difference(set(good_annotations.keys())))
+
     for position in excluded_positions:
         if (experiment_root / position).exists():
             print(f'Found an excluded position to delete {position}')
             if not dry_run:
+                (experiment_root / 'excluded_positions' / position).mkdir(parents=True,exist_ok=True)
```
Comment (zpincus) [inline elegant/clean_timepoint_data.py]: `excluded_root = experiment_root / 'excluded_positions'`; then use that throughout.

File: elegant/clean_timepoint_data.py [hunk 2]
```diff
@@ -113,11 +122,30 @@ def remove_excluded_positions(experiment_root,dry_run=False):
...
+                (experiment_root / 'excluded_positions' / 'annotations').mkdir(parents=True,exist_ok=True)
+
+                shutil.copy(str(experiment_root / position / 'position_metadata.json'),
```
Comment (zpincus) [inline elegant/clean_timepoint_data.py]: As of at least Python 3.6 (if not earlier?) `shutil` can ingest `pathlib.Path` objects directly, without stringifying them. (In at 3.6, it should be a bug for any python standard library function to not be able to deal with `pathlib.Path` objects: [PEP 519](https://www.python.org/dev/peps/pep-0519/).)

File: elegant/clean_timepoint_data.py [hunk 3]
```diff
@@ -113,11 +122,30 @@ def remove_excluded_positions(experiment_root,dry_run=False):
...
+                    str(experiment_root / 'excluded_positions' / position / 'position_metadata.json'))
+                shutil.copy(str(experiment_root / 'annotations' / f'{position}.pickle'),
+                    str(experiment_root / 'excluded_positions' / 'annotations' / f'{position}.pickle'))
+                shutil.copy(str(experiment_root /  'experiment_metadata.json'),
+                    str(experiment_root / 'excluded_positions' / 'experiment_metadata_old.json')) # Back this up JIC....
```
Comment (zpincus) [inline elegant/clean_timepoint_data.py]: why call it `_old` when everything else gets its own filename?

One possibility: if it already exists in the excluded root, try incrementing a counter so you get a save of every version of the file. But that is probably overkill.

**Review by zpincus — CHANGES_REQUESTED** [review]
just a few minor nits / thoughts.

File: elegant/clean_timepoint_data.py (hunk 3 above)
Comment (drew-sinha) [reply elegant/clean_timepoint_data.py]: Deleted the "_old"; leaving it to the user to be careful with handling the experiment_metadata (as also documented in the docstring).

**Comment by drew-sinha** [comment]
Changes should be resolved

File: elegant/clean_timepoint_data.py [hunk 4]
```diff
@@ -104,6 +104,15 @@ def remove_timepoint_from_experiment(experiment_root, timepoint, dry_run=False):
...
     """Deletes excluded positions from an experiment directory
 
+    This function deletes position folders from the specified experiment directory,
+    but saves position_metadata and annotations into the 'excluded_positions' subfolder.
+    'excluded_positions' has a parallel structure to the standard experiment root
+    Each position has a folder containing a copy of its position_metadata;
+    an 'annotations' subfolder contains all of the annotations for excluded positions;
+    and, a copy of the old experiment_metadata lies in this subfolder
+    (though subsequent calls to remove_excluded_positions will overwrite this copy of the experiment metadata).
+    Thus, it is recommended that this function be run on completed experiments to prevent data loss.
```
Comment (zpincus) [inline elegant/clean_timepoint_data.py]: Well, not quite. Running the function twice after excluding more positions will create data loss, even if the experiment was completed prior to either run. But see below for a better idea.

File: elegant/clean_timepoint_data.py [hunk 5]
```diff
@@ -113,11 +122,32 @@ def remove_excluded_positions(experiment_root,dry_run=False):
...
             if not dry_run:
-                shutil.rmtree(str(experiment_root / position))
+                (excluded_root / position).mkdir(parents=True,exist_ok=True)
+                (excluded_root / 'annotations').mkdir(parents=True,exist_ok=True)
+
+                shutil.copy(experiment_root / position / 'position_metadata.json',
+                    excluded_root / position / 'position_metadata.json')
+                shutil.copy(experiment_root / 'annotations' / f'{position}.pickle',
+                    excluded_root / 'annotations' / f'{position}.pickle')
+                shutil.copy(experiment_root /  'experiment_metadata.json',
```
Comment (zpincus) [inline elegant/clean_timepoint_data.py]: What about just not copying it if it's already there? That way the original version is preserved, even if the function gets run several times...

**Review by zpincus — COMMENTED** [review]
A couple stray thoughts

File: elegant/clean_timepoint_data.py (hunk 5 above)
Comment (drew-sinha) [reply elegant/clean_timepoint_data.py]: Good call. Implemented that strategy.

**Comment by zpincus** [comment]
Good catch about the masks too!

**Pull request merged**